vm/src/builtins/union.rs

```diff
@@ -196,21 +196,10 @@ fn dedup_and_flatten_args(args: PyTupleRef, vm: &VirtualMachine) -> PyTupleRef {
     let mut new_args: Vec<PyObjectRef> = Vec::with_capacity(args.len());
     for arg in &args {
         if !new_args.iter().any(|param| {
-            match (
-                PyTypeRef::try_from_object(vm, param.clone()),
-                PyTypeRef::try_from_object(vm, arg.clone()),
-            ) {
-                (Ok(a), Ok(b))
-                    if a.is(vm.ctx.types.generic_alias_type)
-                        && b.is(vm.ctx.types.generic_alias_type) =>
-                {
-                    param
-                        .rich_compare_bool(arg, PyComparisonOp::Eq, vm)
-                        .ok()
-                        .unwrap()
-                }
-                _ => param.is(arg),
-            }
+            param
+                .rich_compare_bool(arg, PyComparisonOp::Eq, vm)
+                .ok()
+                .unwrap()
```

Comment:
For `Result` type, `.ok().unwrap()` equals to `unwrap()`.
But why unwrap? If there is no special guarantee `unwrap` is safe, propagate it. Otherwise use expect with message about the guarantee.

Reply:
Oh, the unwrap part looks like a bug, but it already existed in previous version.

Reply:
I tested again with your feedback, and figured out `ok()` isn't neccessary.
So I made pr #4036 for this.